csharp/src/elements/Geometry/Polygon.cs

```diff
@@ -91,19 +132,163 @@ public bool Contains(Polygon polygon)
+        /// Returns true if the supplied Polygon is contained by or coincides with the perimeter of the supplied Polygon.
         /// </returns>
         public bool Covers(Polygon polygon)
         {
+            var tolerance = 0.00001;
+            var clipper = new Clipper();
+            var solution = new List<List<IntPoint>>();
+            clipper.AddPath(this.ToClipperPath(), PolyType.ptSubject, true);
+            clipper.AddPath(polygon.ToClipperPath(), PolyType.ptClip, true);
+            clipper.Execute(ClipType.ctIntersection, solution);
+            if (solution.Count != 1)
+            {
+                return false;
+            }
+            var testPolygon = solution.First().ToPolygon();
+            if (Math.Abs(polygon.Area - testPolygon.Area) > tolerance)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Tests if this Polygon covers all of the supplied Polygons. This Polygon covers the supplied Polygons if all points of the supplied Polygons fall within or on the perimeter of this Polygon.
+        /// </summary>
+        /// <param name="polygons">The list of Polygons to compare.</param>
+        /// <returns>
+        /// Returns true if any of the supplied Polygons intersects with this Polygon.
+        /// </returns>
+        public bool Covers(IList<Polygon> polygons)
+        {
+            var tolerance = 0.00001;
+            var clipper = new Clipper();
+            var solution = new List<List<IntPoint>>();
+            foreach (Polygon polygon in polygons)
+            {
+                clipper.AddPath(this.ToClipperPath(), PolyType.ptSubject, true);
+                clipper.AddPath(polygon.ToClipperPath(), PolyType.ptClip, true);
+                clipper.Execute(ClipType.ctIntersection, solution);
+                if (solution.Count == 0)
+                {
+                    return false;
+                }
+                var testPolygon = solution.First().ToPolygon();
+                if (Math.Abs(polygon.Area - testPolygon.Area) > tolerance)
+                {
+                    return false;
+                }
+                solution.Clear();
+                clipper.Clear();
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Tests if the supplied 2D Vector3 point is outside the perimeter of this Polygon.
+        /// </summary>
+        /// <param name="point">The 2D Vector3 point to compare.</param>
+        /// <returns>
+        /// Returns true if the supplied 2D Vector3 point falls outside the perimeter of this Polygon.
+        /// </returns>
+        public bool Disjoint(Vector3 point)
+        {
+            var scale = 1024.0;
             var thisPath = this.ToClipperPath();
-            var polyPath = polygon.ToClipperPath();
-            foreach (IntPoint vertex in polyPath)
+            var intPoint = new IntPoint(point.X * scale, point.Y * scale);
```

Comment:
Isn't this section doing a `Contains`? Can we use our new method here?

Reply:
Done, I believe.